app/Http/Controllers/ProfilesController.php

```diff
@@ -15,9 +15,15 @@ class ProfilesController extends Controller
      */
     public function show(User $user)
     {
-        return view('profiles.show', [
+        $data = [
             'profileUser' => $user,
             'activities' => Activity::feed($user)
-        ]);
+        ];
+
+        if (request()->expectsJson()) {
+            return $data;
+        }
+
+        return view('profiles.show', $data);
```

Comment:
The typehint above the function doesn't feel right in my eyes.

Reply:
Yeah! That's better. But, I think I'd like to hear @jeffreyway's thoughts.